src/main/java/net/snowflake/ingest/streaming/internal/BlobBuilder.java

```diff
@@ -135,17 +135,21 @@ static <T> Blob constructBlobAndMetadata(
                     AbstractRowBuffer.buildEpInfoFromStats(
                         serializedChunk.rowCount,
                         serializedChunk.columnEpStatsMapCombined,
-                        internalParameterProvider.setAllDefaultValuesInEp()))
+                        internalParameterProvider.setAllDefaultValuesInEp(),
+                        internalParameterProvider.isEnableDistinctValuesCount()))
                 .setFirstInsertTimeInMs(serializedChunk.chunkMinMaxInsertTimeInMs.getFirst())
                 .setLastInsertTimeInMs(serializedChunk.chunkMinMaxInsertTimeInMs.getSecond());
 
         if (internalParameterProvider.setIcebergSpecificFieldsInEp()) {
+          final long metadataSize = getParquetFooterSize(compressedChunkData);
```

Comment:
if BDEC wants to set parquet footer size, this logic won't work since you're going off the encrypted byte array.
What we really want is to run this off the unencrypted, unpadded byte array, no?

Lets add a comment here for now that this calculation cannot be used when encryption is on. Also add an assert for internalParameterProvider.getEnableChunkEncryption() == false in here.

Reply:
Added assertion.